### ISSUE TITLE
Additional printer features

### ISSUE THREAD

**Issue by Lunatrius** [Lunatrius/Schematica#102]
All features with a low priority regarding the printer should be suggested here. These might be added to the mod or released as an addon for the mod.
* [ ] Automated printer
  * [ ] Walking AI #48 #91
  * [ ] Refill AI #91
  * [ ] Scaffolding AI #91
  * [ ] Move around camera (smooth or instant?) #48
* [ ] Different placement modes
  * [x] Default XZY
  * [ ] Spherical from player
  * [ ] Spherical from cursor #50

**Comment by MokahTGS** [comment]
Print whole schematic?

**Comment by Lunatrius** [comment]
That's not a *printer* feature that is possible with client side only. An import feature was planned but is currently on hold due to rather limited free time.

**Comment by Epicarseus** [comment]
Would it be possible to copy noteblock data values into a schematic? If so, the printer could try to right-click the noteblock upon placement.

**Comment by Lunatrius** [comment]
It's been often requested, but the client never knows which note the noteblock is actually using. The pitch is never synchronized to the client.

**Comment by ThePlasmaRailgun** [comment]
Maybe you could try and have your mod read what sound file the server was sending you for that noteblock.

**Comment by Pokechu22** [comment]
Actually, 1.13 (specifically, via [the flattening](https://minecraft.gamepedia.com/1.13/Flattening)), note blocks will lose the block entity and will instead have a block state, so it'd be much easier there (block states are synced with the client).  (But the [block action](http://wiki.vg/Block_Actions#Note_Block) also is present so you can get the pitch from it)

**Comment by Lunatrius** [comment]
Is the block action available on the client though?

Looks like this is something that can be definitely fixed in 1.13 with flattening (it's going to break quite a few convenience commands though).

**Comment by Pokechu22** [comment]
Yep.  There's a packet that sends it from the server to the client.  I currently use it to save the noteblock's pitch in World Downloader.  Of course, if you're actively changing the pitch, then things get harder because of latency - WDL only needs to passively get the pitch; schematica would need to change it until it sounds right which might be hard.

**Comment by Lunatrius** [comment]
Oh, so it's what I already considered as an option. Wouldn't work when saving schematics either.

**Comment by Erumeldor** [comment]
I would really love to see an option to an more automated printer. Regarding several anti-cheat-plugins, a block has to be placed while looking at it.
Also, blocks that can't be seen are placed by Schematica, and that's against the regulations of these anti-cheat-plugins too.
I think, first step would be to roatete the camera to the block you are looking at, even if mooving around with that function would be horrible.
